server/sequencer_manager.go

```diff
@@ -54,42 +55,55 @@ func (s SequencerManager) ExecutePass(logIDs []int64, logctx LogOperationManager
 	successCount := 0
 	leavesAdded := 0
 
+	var wg sync.WaitGroup
+
 	for _, logID := range logIDs {
-		// See if it's time to quit
-		select {
-		case <-logctx.ctx.Done():
-			return true
-		default:
-		}
-
-		// TODO(Martin2112): Honor the sequencing enabled in log parameters, needs an API change
-		// so deferring it
-		storage, err := s.registry.GetLogStorage()
-		if err != nil {
-			glog.Warningf("%v: failed to acquire log storage: %v", logID, err)
-			continue
-		}
-		ctx := util.NewLogContext(logctx.ctx, logID)
-
-		// TODO(Martin2112): Allow for different tree hashers to be used by different logs
-		hasher, err := merkle.Factory(merkle.RFC6962SHA256Type)
-		if err != nil {
-			glog.Errorf("Unknown hash strategy for log %d: %v", logID, err)
-			continue
-		}
-		sequencer := log.NewSequencer(hasher, logctx.timeSource, storage, s.keyManager)
-		sequencer.SetGuardWindow(s.guardWindow)
-
-		leaves, err := sequencer.SequenceBatch(ctx, logID, logctx.batchSize)
-		if err != nil {
-			glog.Warningf("%v: Error trying to sequence batch for: %v", logID, err)
-			continue
-		}
-
-		successCount++
-		leavesAdded += leaves
+		wg.Add(1)
+		logID := logID
+		go func() {
+			defer wg.Done()
+
+			start := time.Now()
+
+			// TODO(Martin2112): Honor the sequencing enabled in log parameters, needs an API change
+			// so deferring it
+			storage, err := s.registry.GetLogStorage()
```

Comment:
Done, I left `hasher` and `sequencer` inside one depends on the other and, in theory at least, `hasher` could be different for each tree.